RedProtect-Spigot/src/main/java/br/net/fabiozumbi12/RedProtect/Bukkit/listeners/RPCommands.java

```diff
@@ -2592,13 +2594,20 @@ private static void handleRename(Player p, String newName) {
             }
 
             //filter region name
-            newName = newName.replace(" ", "_").replaceAll("[^\\p{L}_0-9 ]", "");
+            newName = newName.replace(" ", "_");
+
+            if (hasSpecialChar(newName)) {
+                p.sendMessage(RPLang.get("_redprotect.prefix") + " RedProtect name cannot has special character!");

```

Comment:
Please use `RPLang.sendMessage(Player p, String key)` instead of `p.sendMessage(String message)`.